examples/blur/main.go

```diff
@@ -49,17 +49,33 @@ func (g *Game) Draw(screen *ebiten.Image) {
 
 	// Box blur (7x7)
 	// https://en.wikipedia.org/wiki/Box_blur
+	//
+	// Note that this is a fixed function implementation of a box blur - more
+	// efficiency can be gained by using a separable blur
+	// (blurring horizontally and vertically separately, or for large blurs,
+	// even multiple horizontal or vertical passes), ideally combined with
+	// doing the summing up in a fragment shader (Kage can be used here).
+	//
+	// So this implementation only serves to demonstrate use of alpha blending.
+	layers := 0
 	for j := -3; j <= 3; j++ {
 		for i := -3; i <= 3; i++ {
 			op := &ebiten.DrawImageOptions{}
 			op.GeoM.Translate(float64(i), 244+float64(j))
-			// Alpha scale should be 1.0/49.0, but accumulating 1/49 49 times doesn't reach to 1, because
-			// the final color is affected by the destination alpha when CompositeModeSourceOver is used.
+			// Note that the final color is affected by the destination alpha when CompositeModeSourceOver is used.
 			// This composite mode is the default mode. See how this is calculated at the doc:
 			// https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2#CompositeMode
 			//
-			// Use a higher value than 1.0/49.0. Here, 1.0/25.0 here to get a reasonable result.
-			op.ColorM.Scale(1, 1, 1, 1.0/25.0)
+			// So if using the same alpha every time, the end result will sure be biased towards the last layer.
+			//
+			// Correct averaging works based on
+			//   Let A_n := (a_1 + ... + a_n) / n
+			//   A_{n+1} = (a_1 + ... + a_{n+1}) / (n + 1)
+			//   A_{n+1} = (n * A_n + a_{n+1)) / (n + 1)
+			//   A_{n+1} = A_n * (1 - 1/(n+1)) + a_{n+1} * 1/(n+1)
```

Comment:
No, because this is alpha blending. Note how we set unpremultiplied colors in those matrices, so it behaves like GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA.

Reply:
Which means, another way to fix this example would be to make it use additive blending (CompositeModeLighter) and the same alpha in every step. Didn't do this as I thought the comment indicates this is an example of using CompositeModeOver.

Plus, this was is more accurate for resulting brightness than using CompositeModeLighter as roundoff errors are compensated by the next pass.

Reply:
Acknowledged. I'd be happy if you could leave a comment about this: the default composite mode works well.

Reply:
Done.